src/main/java/com/vesta/controller/SubjectController.java

```diff
@@ -33,4 +35,22 @@
     @ApiOperation(value = "Returns the all subjects")
     @GetMapping("/subjects/all")
     List<SubjectView> getAll();
+
+    @ApiOperation(value = "Returns the all subjects by FloorID")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Get subjects by FlooID has succeeded"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "Floor not found")
+    })
+    @GetMapping("/subjects/all/by/floor")
```

Comment:
Change the path "/subjects/all/by/floor" to "/subjects/floor/{floorId}" where {floorId} is PathVariable